### PR TITLE
add iso-4 format

### LINKED ISSUE #3: Support PIN block ISO-4 (ISO 9564-1: 2017)
ISO-4 (Format 4)
ISO 9564-1: 2017 Format 4. Format 4 uses AES-128 ECB encryption. PIN Block is padded with fill digit ‘A’ and also with random values from 0 to 15, and the first nibble (which identifies the block format) has the value 4.

[Implementing ISO Format 4 PIN Blocks](https://listings.pcisecuritystandards.org/documents/Implementing_ISO_Format_4_PIN_Blocks_Information_Supplement.pdf)

### PR DESCRIPTION
**Pull request by alovak**
Add iso-4 format according to the following sources:
* https://listings.pcisecuritystandards.org/documents/Implementing_ISO_Format_4_PIN_Blocks_Information_Supplement.pdf
* https://www.eftlab.com/knowledge-base/complete-list-of-pin-blocks

closes #3

### CODE REVIEW

File: encryption/aes_ecb_test.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	})
+
+	t.Run("Create with wrong key", func(t *testing.T) {
+		// short key
+		_, err := NewAesECB([]byte("1234567890"))
+		require.Error(t, err)
+
+		// long key
+		_, err = NewAesECB([]byte("12345678901234561234567890123456"))
+		require.Error(t, err)
```
Comment (adamdecaf) [inline encryption/aes_ecb_test.go:32]: Can we use `.ErrorContains` to verify the exact error we expect?

File: encryption/aes_ecb_test.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestAesECB(t *testing.T) {
+	t.Run("Encode/Decode", func(t *testing.T) {
+		cipher, err := NewAesECB([]byte("1234567890123456"))
```
Comment (adamdecaf) [inline encryption/aes_ecb_test.go:11]: Are we always going be reading ASCII alphanumeric characters?

File: lint-project.sh [hunk 3]
```diff
@@ -0,0 +1,355 @@
+#!/bin/bash
```
Comment (adamdecaf) [inline lint-project.sh:1]: This should be in the .gitignore

File: encryption/aes_ecb_test.go (hunk 2 above)
Comment (alovak) [reply encryption/aes_ecb_test.go:11]: This is just a random key, so it can be anything here.

File: lint-project.sh (hunk 3 above)
Comment (alovak) [reply lint-project.sh:1]: ups.

File: encryption/aes_ecb_test.go (hunk 1 above)
Comment (alovak) [reply encryption/aes_ecb_test.go]: updated

File: .github/workflows/go.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    - name: Install
+      run: make install
+
+    - name: Check
+      if: runner.os == 'Linux'
+      run: make check
+
+    - name: Go Tests
+      if: runner.os != 'Linux'
```
Comment (adamdecaf) [inline .github/workflows/go.yml:40]:
```suggestion
      if: runner.os != 'Windows'
      run: make check

    - name: Go Tests
      if: runner.os == 'Windows'
```

File: encryption/aes_ecb_test.go (hunk 2 above)
Comment (adamdecaf) [reply encryption/aes_ecb_test.go:11]: Can we add a test that doesn't have alphanumeric characters then?

File: encryption/noop.go [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+func NewNoOp() *NoOp {
+	return &NoOp{}
+}
+
+func (n *NoOp) Encrypt(plaintext []byte) ([]byte, error) {
+	return plaintext, nil
+}
+
+func (n *NoOp) Decrypt(ciphertext []byte) ([]byte, error) {
+	return ciphertext, nil
```
Comment (adamdecaf) [inline encryption/noop.go:14]: Do we want to validate lengths still?

File: encryption/aes_ecb_test.go (hunk 2 above)
Comment (alovak) [reply encryption/aes_ecb_test.go]: I replaced it with randomly generated key

**Review by adamdecaf — APPROVED** [review]

**Pull request merged**